### DISCUSSION TITLE
How to identify a connection from the internet properly?

### DISCUSSION THREAD

**Discussion by googollee** [jellyfin/jellyfin#13239] [General] [answered]
Hi. I'm running a self-host Jellyfin server with a traefix server as a reverse proxy. I use Cloudflare CDN to access the Jellyfin server from the internet. When I access the server through the internet, Jellyfin always shows logs like:

```
[INF] [119] Jellyfin.Api.Helpers.MediaInfoHelper: RemoteClientBitrateLimit: 1500000, RemoteIP: 10.42.0.1, IsInLocalNetwork: True
```

The remote ip is not right. It's the IP of my traefik server in the local network, but not a real remote ip. I believe I set traefik server properly to forward the client IP with `X-Real-Ip` and `X-Forward-For`.

I digged into the code. I see Jellyfin retrieves remote IP with [`GetNormalizedRemoteIP()`](https://github.com/jellyfin/jellyfin/blob/55f5eaf0e313783a8e45fa3f3d63a8aea5d91f48/MediaBrowser.Common/Extensions/HttpContextExtensions.cs#L28), which delegates to `context.Connection.RemoteIpAddress`. From [the doc](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.connectioninfo.remoteipaddress?view=aspnetcore-9.0), it doesn't return a HTTP remote IP, but a TCP connection remote IP. It will be always the reverse proxy IP as the TCP connection is created between proxy and Jellyfin server.

Is it the intended behavior? To get a HTTP client IP, `GetNormalizedRemoteIP()` should read from request headers with `X-Real-Ip` or `X-Forward-For`.

**Comment by crobibero** [comment, marked as answer]
https://jellyfin.org/docs/general/networking/#known-proxies

**Reply by googollee** [reply]
Thank you.
